Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift

```diff
@@ -233,7 +233,7 @@ extension NIOAsyncSequenceProducer {
             @inlinable
             deinit {
                 // We need to call finish here to resume any suspended continuation.
-                self._throwingSource.finish()
+                //self._throwingSource.finish()
```

Comment:
Should this make use of `self.finishOnDeinit`?

Reply:
yes! all still WIP. hardest part will be getting all the tests to work etc :)

Reply:
Ack. Apologies. Didn't mean to jump the gun :)